videozen-frontend/src/components/Videos/Videos.jsx

```diff
@@ -0,0 +1,62 @@
+import { useData } from "../../context/data-context";
+import { Link } from "react-router-dom";
+
+import "./Videos.css";
+
+import Video from "../Video/Video";
+import { useState } from "react";
+
+const Videos = () => {
+  const { state, dispatch } = useData();
+  const [searchedText, setSearchedText] = useState("");
+
+  let allVideos = state.allPlaylists.reduce(
+    (accumulator, item) => [...accumulator, ...item.videos],
+    []
+  );
+
+  let searchedVideos = allVideos.filter((video) =>
+    video.title.toLowerCase().includes(searchedText.toLowerCase())
+  );
+
+  return (
+    <div className="videos">
+      <input
+        type="text"
+        value={searchedText}
+        className="form-control"
+        placeholder="Search for videos by title..."
+        onChange={(e) => setSearchedText(e.target.value)}
+      />
+      {searchedVideos.length === 0 && (
+        <h3>
+          No Videos found with{" "}
+          <em>
+            <q>{searchedText}</q>
+          </em>
+        </h3>
+      )}
+      <div className="card-row">
+        {searchedVideos.map((video) => (
+          <Link
+            to={`/playlist/allvideos/${video.id}`}
+            state={video}
+            style={{ textDecoration: "none", color: "black" }}
+            onClick={() =>
+              dispatch({ type: "ADD_TO_HISTORY", payload: { video: video } })
```

Comment:
can use only one video will do
```suggestion
              dispatch({ type: "ADD_TO_HISTORY", payload: { video } })
```